seed_isort_config.py

```diff
@@ -53,26 +58,40 @@ def main(argv=None):
 
     third_party = ','.join(sorted(third_party_imports(filenames)))
 
-    for filename in ('.isort.cfg', 'setup.cfg', 'tox.ini'):
+    existing_conf_filename = None
+    for filename, config_start_re in CONFIG_FILE_TO_START_RE.items():
         if not os.path.exists(filename):
             continue
 
+        existing_conf_filename = filename
+
         with io.open(filename, encoding='UTF-8') as f:
             contents = f.read()
 
+        if not config_start_re.search(contents):
+            continue
+
         if THIRD_PARTY_RE.search(contents):
             replacement = r'known_third_party\1=\2{}'.format(third_party)
-            contents = THIRD_PARTY_RE.sub(replacement, contents)
-            with io.open(filename, 'w', encoding='UTF-8') as f:
-                f.write(contents)
+            updated_contents = THIRD_PARTY_RE.sub(replacement, contents)
+        else:
+            replacement = r'\1\nknown_third_party={}'.format(third_party)
+            updated_contents = config_start_re.sub(replacement, contents)
+        with io.open(filename, 'w', encoding='UTF-8') as f:
+            f.write(updated_contents)
             break
     else:
-        print(
-            'Could not find a `known_third_party` setting in any of '
-            '.isort.cfg, setup.cfg, tox.ini.  '
-            'Set up an initial config and run again!',
-        )
-        return 1
+        conf_filename = existing_conf_filename or '.isort.cfg'
+        conf_start = CONFIG_FILE_TO_START_RE[conf_filename]
+
+        print('Adding isort config to {}...'.format(conf_filename))
+
+        with io.open(conf_filename, 'a', encoding='UTF-8') as f:
+            f.write('{}{}\n{}'.format(
+                '\n' if existing_conf_filename else '',
+                conf_start.pattern.replace('\\', '')[2:-2],
```

Comment:
yikes this is fragile.  iirc patterns aren't guaranteed to be verbatim input (or even present in some implementations) -- hopefully this goes away with the suggestions above?

Reply:
Will give it a shot!